src/sentry/api/endpoints/auth_index.py

```diff
@@ -96,10 +110,7 @@ def _validate_superuser(self, validator, request):
         validator.is_valid()
         authenticated = None
 
-        if (
-            request.user.has_usable_password()
-            or Authenticator.objects.filter(user_id=request.user.id, type=3).exists()
-        ):
+        if self._require_password_or_u2f_check(request):
```

Comment:
The helper function doesn't need to be a static method. It's meaningful only in the case of validating a superuser, so we can define it nested in the method, i.e.,

```python
    def _validate_superuser(self, validator, request):
        validator.is_valid()
        authenticated = None

        def require_password_or_u2f_check():
            return ...

        if require_password_or_u2f_check():
            ...
```

This is preferable because the "is_self_hosted" part isn't really applicable to the "require a password or U2F check" question except when it's being called from `_validate_superuser`. Someone trying to call the same static method from a new place could get seriously confused.

As a bonus, you can refer directly to `request` and other variables without passing them as arguments.

The helper function is also an opportunity to explode the boolean logic, so we can explain each part individually. Such as:
```python
    def require_password_or_u2f_check():
        if is_self_hosted():
            # Don't need to check because...
            return False
        if request.user.has_usable_password():
            # etc.
```